aicsimageio/tests/writers/test_ome_tiff_writer.py

```diff
@@ -1,80 +1,111 @@
+    with pytest.raises(Exception):
+        writer = OmeTiffWriter(resources_dir / filename, overwrite_file=True)
+        writer.save(image_to_save)
 
-    def test_loadAssertionError(self):
-        image_to_save = np.ones((1, 2, 3, 4, 5, 6))
-        with self.assertRaises(Exception):
-            self.writer.save(image_to_save)
 
+def test_overwriteFile(resources_dir):
     """
     Test to check if save() can overwrite a file
     """
+    with OmeTiffWriter(resources_dir / filename, overwrite_file=True) as writer:
+        writer.save(image)
 
-    def test_overwriteFile(self):
-        with OmeTiffWriter(self.file, overwrite_file=True) as writer:
-            writer.save(self.image)
 
+def test_dontOverwriteFile(resources_dir):
     """
     Test to check if save() will raise error when user does not want to overwrite a file that exists
     """
+    with pytest.raises(Exception):
+        with OmeTiffWriter(resources_dir / filename) as writer:
+            writer.save(image)
 
-    def test_dontOverwriteFile(self):
-        with self.assertRaises(Exception):
-            with OmeTiffWriter(self.file) as writer:
-                writer.save(self.image)
 
+def test_noopOverwriteFile(resources_dir):
     """
     Test to check if save() silently no-ops when user does not want to overwrite exiting file
     """
-
-    def test_noopOverwriteFile(self):
-        with open(self.file, "w") as f:
-            f.write("test")
-        with OmeTiffWriter(self.file, overwrite_file=False) as writer:
-            writer.save(self.image)
-        with open(self.file, "r") as f:
-            line = f.readline().strip()
-            self.assertEqual("test", line)
-
-    def test_big_tiff(self):
-        x = np.zeros((10, 10))
-        assert OmeTiffWriter._size_of_ndarray(data=x) == 10*10*x.itemsize
+    with open(resources_dir / filename, "w") as f:
+        f.write("test")
+    with OmeTiffWriter(resources_dir / filename, overwrite_file=False) as writer:
+        writer.save(image)
+    with open(resources_dir / filename, "r") as f:
+        line = f.readline().strip()
+        assert "test" == line
+    # os.remove(resources_dir / filename)
+
+
+def test_big_tiff():
+    x = np.zeros((10, 10))
+    assert OmeTiffWriter._size_of_ndarray(data=x) == 10*10*x.itemsize
+
+
+@pytest.mark.parametrize("dims, expected_t, expected_c, expected_z, expected_y, expected_x", [
+    ("TCZYX", 1, 40, 3, 128, 256),
+    ("TZCYX", 1, 3, 40, 128, 256),
+    ("CZTYX", 3, 1, 40, 128, 256),
+    ("CTZYX", 40, 1, 3, 128, 256),
+    ("ZCTYX", 3, 40, 1, 128, 256),
+    ("ZTCYX", 40, 3, 1, 128, 256),
+    ("STCZYX", 1, 40, 3, 128, 256),
+    (pytest.param("XY", 1, 40, 3, 128, 256, marks=pytest.mark.raises(exception=AssertionError))),
+    (pytest.param("ABCD", 1, 40, 3, 128, 256, marks=pytest.mark.raises(exception=AssertionError))),
+    (pytest.param("XXXXX", 1, 40, 3, 128, 256, marks=pytest.mark.raises(exception=AssertionError))),
```

Comment:
Could you add comments of "This raises because ..."

Reply:
I can, but is it not enough to have the assertion strings?

Reply:
Yea I made this comment before I saw those. All good.

Reply:
I made the change anyway to remove the asserts and have it raise InvalidDimensionOrderingErrors instead.